data/json/mapgen/lab/lab.json

```diff
@@ -0,0 +1,956 @@
+        "|-|--|+--|-D^----|-----|",
+        "|.rrr..........rrrrrr..|",
+        "|......................|",
+        "|......................|",
+        "|r..rr..rr....rr..rr..r|",
+        "|r..rr..rr....rr..rr..r|",
+        "|r..rr..rr....rr..rr..r|",
+        "|r..rr..rr....rr..rr..r|",
+        "|r..rr..rr....rr..rr..r|",
+        "|r....................r|",
+        "|----------------------|"
+      ],
+      "palettes": [
+        "lab_palette"
+      ],
+      "place_monsters": [ { "monster": "GROUP_MAYBE_MIL", "x": [ 1, 22 ], "y": [ 1, 22 ], "density": 0.1 } ],
+      "place_monster": [
+        { "monster": "mon_zombie_bio_op", "x": [ 1, 22 ], "y": [ 1, 22 ], "chance": 33 },
+        { "monster": "mon_zombie_hulk", "x": [ 11, 12 ], "y": [ 6, 22 ], "chance": 90 }
+      ],
+      "furniture": {
+        "^": "f_rubble"
+      },
+      "terrain": {
+        "^": "t_rock_floor"
+      },
+      "place_items": [
+        { "item": "building_rubble", "x": 11, "y": 4, "chance": 100 },
+        { "item": "building_rubble", "x": 12, "y": 13, "chance": 100 }
+      ],
+      "mapping": {
+        "l": {
+          "items": [
+            { "item": "clothing_soldier_set", "chance": 100 },
+            { "item": "gear_soldier_sidearm", "chance": 33},
+            { "item": "book_military", "chance": 33}
+          ]
+        },
+        "b": {
+          "items": [ { "item": "bed", "chance": 50 } ]
+        },
+        "d": {
+          "items": [ { "item": "office", "chance": 50 } ]
+        },
+        "^": {
+          "items": [ { "item": "building_rubble", "chance": 100, "repeat": 2 } ]
+        },
+        "r": {
+          "items": [
+            { "item": "military", "chance": 10 },
+            { "item": "ammo_milspec", "chance": 10},
+            { "item": "mags_milspec", "chance": 10},
+            { "item": "guns_milspec", "chance": 10}
+          ]
+        }
+      }
+    }
+  },
+  {
+    "type": "mapgen",
+    "method": "json",
+    "nested_mapgen_id": "lab_border_walls",
+    "object": {
+      "mapgensize": [ 24, 24 ],
+      "place_nested": [
+        { "chunks": [ "sub_t_concrete_wall" ], "x": 11, "y": 23, "neighbors": { "south": ["empty_rock"] } },
+        { "chunks": [ "sub_t_concrete_wall" ], "x": 12, "y": 23, "neighbors": { "south": ["empty_rock"] } },
+        { "chunks": [ "sub_t_concrete_wall" ], "x": 23, "y": 11, "neighbors": { "east": ["empty_rock"] } },
+        { "chunks": [ "sub_t_concrete_wall" ], "x": 23, "y": 12, "neighbors": { "east": ["empty_rock"] } },
+        { "chunks": [ "lab_north_wall" ], "x": 0, "y": 0, "neighbors": { "north": ["empty_rock"] } },
+        { "chunks": [ "lab_west_wall" ], "x": 0, "y": 0, "neighbors": { "west": ["empty_rock"] } }
+      ]
+    }
+  },
+  {
+    "type": "mapgen",
+    "method": "json",
+    "nested_mapgen_id": "lab_north_wall",
+    "object": {
+      "mapgensize": [ 24, 24 ],
+      "place_terrain": [
+        { "ter": "t_concrete_wall", "x": 0, "y": 0 },
```

Comment:
I believe you can draw lines of terrain:

https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/MAPGEN.md#22-set

Reply:
Am i wrong that there is an error in the example of "line" in MAPGEN.md?
`Example: { "point": "terrain", "id": "t_lava", "x:" 5, "y": 5, "x2": 20, "y2": 20 }`

Reply:
Yes, docs can be outdated a bit.